src/main/kotlin/com/esops/service/OrderService.kt

```diff
@@ -3,25 +3,24 @@ package com.esops.service
 import com.esops.configuration.PlatformFeesConfiguration
 import com.esops.entity.*
 import com.esops.model.AddOrderRequestBody
+import com.esops.repository.OrderRepository
 import jakarta.inject.Singleton
 import java.math.BigInteger
-import java.util.*
 
 @Singleton
-class OrderService(private var userService: UserService,
-                   private var platformService: PlatformService,
-                   private var platformFeesConfiguration: PlatformFeesConfiguration) {
+class OrderService(
+    private var userService: UserService,
+    private var platformService: PlatformService,
+    private var platformFeesConfiguration: PlatformFeesConfiguration,
+    private var orderRepository: OrderRepository
+) {
 
-    private var orders = HashMap<String, HashMap<Long, Order>>()
-    private var buyOrderQueue = PriorityQueue(BuyOrderComparator)
-    private var sellOrderQueue = PriorityQueue(SellOrderComparator)
-    private var orderIDCounter: Long = 0
 
     fun placeOrder(username: String, addOrderRequestBody: AddOrderRequestBody): Order {
         userService.checkOrderPlacement(username, addOrderRequestBody)
-        orderIDCounter++
+        orderRepository.incrementOrderIDCounter()
         val user = this.userService.getUser(username)
```

Comment:
NONE OF THIS SHOULD HAPPEN IN SERVICE. If anything is store related, you have a repo, handle it there

Reply:
MovedorderID increment part in repository